controllers/pravega_segmentstore.go

```diff
@@ -640,8 +644,17 @@ func MakeSegmentStoreExternalServices(p *api.PravegaCluster) []*corev1.Service {
 			service.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeLocal
 		}
 		if p.Spec.Pravega.SegmentStoreLoadBalancerIP != "" {
-			service.Spec.Ports[0].Port = int32(serviceport) + i
-			service.Spec.Ports[1].Port = int32(adminPort) + i
+			for util.ContainsElement(p.Spec.ReservedPortList, int32(serviceport)+i+servicePortIncrement) {
+				servicePortIncrement++
+			}
+
+			service.Spec.Ports[0].Port = int32(serviceport) + i + servicePortIncrement
+
+			for util.ContainsElement(p.Spec.ReservedPortList, int32(adminPort)+i+adminPortIncrement) {
+				adminPortIncrement++
+			}
+			service.Spec.Ports[1].Port = int32(adminPort) + i + adminPortIncrement
+
```

Comment:
I'm not sure why we have all this math here.
We could have a `currentServicePort` initialized to `int32(serviceport)` and a `currentAdminPort` initialized to `int32(adminPort)`. 
At the end of every iteration of SS replicas we run `currentServicePort++` and `currentAdminPort++`.
 If `currentServicePort` value is on reserved list, we run `currentServicePort++`.
Again, if `currentAdminPort` value is on reserved list, we run  `currentAdminPort++`

Reply:
Thanks @jkhalack for the feedback, have made changes as suggested.